### ISSUE TITLE
Warp doesn't respect `$XDG_CONFIG_HOME`

### ISSUE THREAD

**Issue by askoufis** [warpdotdev/Warp#814]
### Describe the solution you'd like?

```text
Warp should ideally respect the `XDG_CONFIG_HOME` environment variable, as specified [here](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html).
```

### Is your feature request related to a problem? Please describe.

```text
I (and I assume lots of other people) prefer to store config somewhere other than my home directory, for example `~/.config`. As of version `v0.2022.03.14.08.49.stable_00`, keybindings are now stored in `~/.warp/keybindings.yaml`. If a user has configured `XDG_CONFIG_HOME`, this file should be stored in `warp/keybindings.yaml` within that folder.
```

**Comment by SebDanielsson** [comment]
Yes please make this happen. I don't want my home dir cluttered with app config.

For example custom themes should be stored at:
~/.config/warp/themes/themename.yaml

**Comment by mrolli** [comment]
This is a must have for me. I mean, this is a no-brainer to implement, but a showstopper for me.

**Comment by mrolli** [comment]
> Wow same, I enjoy warp, but it's hard to use applications that don't respect XDG specs in 2022

And especially if you intend to land on Linux, where respecting XDG is a common thing nowadays. Even on macOS, a lot of apps stopped polluting $HOME and are respecting XDG, kitty, alacritty, ... just to name some other popular Terminal alternatives.

Actually one might even argue, that using $HOME on macOS is the badest choice. Either honour XDG or use the macOS default, which would be ~/Library/Application Support/$APP/. 

But as I said, it's so easy to implement:

`config_home="${XDG_CONFIG_HOME:-$HOME}/warp"`

Done, backwards compatible.
And the same for the other $XDG_* environment variables.

I can't understand why this ticket is lingering around that long already.

**Comment by pauliesnug** [comment]
bumping

**Comment by aaronmallen** [comment]
@warpdotdev-devx how is this issue 3 years old!?!?

**Comment by jtbrough** [comment]
I hope @warpdotdev-devx can find time to adopt this rational standard.

**Comment by tsdevau** [comment]
> Wow same, I enjoy warp, but it's hard to use applications that don't respect XDG specs in 2022

What about one that still doesn't in late 2025?!?
